src/base/utils/string.cpp

```diff
@@ -190,11 +190,14 @@ QString Utils::String::wildcardToRegex(const QString &pattern)
     return qt_regexp_toCanonical(pattern, QRegExp::Wildcard);
 }
 
-bool Utils::String::parseBool(const QString &string, const bool defaultValue)
+std::optional<bool> Utils::String::parseBool(const QString &string)
 {
-    if (defaultValue)
-        return (string.compare("false", Qt::CaseInsensitive) == 0) ? false : true;
-    return (string.compare("true", Qt::CaseInsensitive) == 0) ? true : false;
+    if (string.compare("false", Qt::CaseInsensitive) == 0)
+        return false;
+    if (string.compare("true", Qt::CaseInsensitive) == 0)
```

Comment:
I would make this the first comparison as "true" string is shorter than the other.